packages/core/src/App/Containers/Layout/header/dtrader-header-wallets.tsx

```diff
@@ -0,0 +1,142 @@
+    const { is_bot_allowed, is_logged_in, is_mt5_allowed, is_dxtrade_allowed } = client;
+    const { app_routing_history, current_language } = common;
+    const { header_extension } = ui;
+    const { setTogglePlatformType } = traders_hub;
+
+    const filterPlatformsForClients = (payload: typeof platform_config) =>
+        payload.filter(config => {
+            if (config.link_to === routes.mt5) {
+                return !is_logged_in || is_mt5_allowed;
+            }
+            if (config.link_to === routes.dxtrade) {
+                return is_dxtrade_allowed;
+            }
+            if (
+                config.link_to === routes.bot ||
+                config.href === routes.binarybot ||
+                config.href === routes.smarttrader
+            ) {
+                return is_bot_allowed;
+            }
+            return true;
+        });
+
+    return (
+        <div className='header__menu-left'>
+            <DesktopWrapper>
+                <PlatformSwitcher
+                    app_routing_history={app_routing_history}
+                    platform_config={filterPlatformsForClients(platform_config)}
+                    setTogglePlatformType={setTogglePlatformType}
+                    current_language={current_language}
+                />
+            </DesktopWrapper>
+            <MobileWrapper>
+                <ToggleMenuDrawer platform_config={filterPlatformsForClients(platform_config)} />
+                {header_extension && is_logged_in && (
+                    <div className='header__menu-left-extensions'>{header_extension}</div>
+                )}
+            </MobileWrapper>
+            <DesktopWrapper>
+                <TradersHubHomeButton />
+            </DesktopWrapper>
+            <MenuLinks />
+        </div>
+    );
+});
+
+const MenuRight = observer(() => {
+    const { client, ui } = useStore();
+    const { is_logged_in, is_logging_in, is_switching } = client;
+    const { is_mobile } = ui;
+
+    const { is_in_progress } = useWalletMigration();
+    const active_account = useActiveAccount();
+    const currency = active_account?.currency ?? '';
+
+    return (
+        <div
+            className={classNames('header__menu-right', {
+                'header__menu-right--hidden': is_mobile && is_logging_in,
+            })}
+        >
+            <DesktopWrapper>
+                <div className='header__menu--dtrader--separator--account'>
+                    <Divider />
+                </div>
+            </DesktopWrapper>
+            {(is_logging_in || is_switching) && (
+                <div
+                    id='dt_core_header_acc-info-preloader'
+                    className={classNames('acc-info__preloader__dtrader acc-info__preloader__dtrader--wallets', {
+                        'acc-info__preloader__dtrader--no-currency': !currency,
+                        'acc-info__preloader__dtrader--is-crypto': getDecimalPlaces(currency) > 2,
```

Comment:
```suggestion
                        'acc-info__preloader__dtrader--is-crypto': active_account?.currency_config?.fractional_digits > 2,
```
Let's try not to use legacy code as much as we can 🙏🏻